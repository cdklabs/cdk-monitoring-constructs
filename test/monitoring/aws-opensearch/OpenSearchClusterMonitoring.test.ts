import { Duration, Stack } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import * as elasticsearch from "aws-cdk-lib/aws-elasticsearch";
import * as opensearch from "aws-cdk-lib/aws-opensearchservice";

import {
  AlarmWithAnnotation,
  OpenSearchClusterMonitoring,
  OpenSearchClusterStatus,
} from "../../../lib";
import { TestMonitoringScope } from "../TestMonitoringScope";

const stack = new Stack();
[
  new elasticsearch.CfnDomain(stack, "ElasticsearchCfnDomain", {
    domainName: "es-cfn-domain",
  }),
  new elasticsearch.Domain(stack, "ElasticsearchDomain", {
    domainName: "es-domain",
    version: elasticsearch.ElasticsearchVersion.V7_7,
  }),
  new opensearch.CfnDomain(stack, "OpenSearchCfnDomain", {
    domainName: "os-cfn-domain",
  }),
  new opensearch.Domain(stack, "OpenSearchDomain", {
    domainName: "os-domain",
    version: opensearch.EngineVersion.OPENSEARCH_1_0,
  }),
].forEach((domain) => {
  test(`snapshot test: no alarms for ${domain.node.id}`, () => {
    const scope = new TestMonitoringScope(
      stack,
      "Scope-NoAlarms-" + domain.node.id,
    );

    new OpenSearchClusterMonitoring(scope, {
      domain,
    });

    expect(Template.fromStack(stack)).toMatchSnapshot();
  });

  test(`snapshot test: all alarms for ${domain.node.id}`, () => {
    const scope = new TestMonitoringScope(
      stack,
      "Scope-AllAlarms-" + domain.node.id,
    );

    let numAlarmsCreated = 0;

    new OpenSearchClusterMonitoring(scope, {
      domain,
      addIndexingLatencyP50Alarm: {
        Warning: {
          maxLatency: Duration.millis(500),
          evaluationPeriods: 10,
        },
      },
      addIndexingLatencyP90Alarm: {
        Warning: {
          maxLatency: Duration.millis(600),
        },
      },
      addIndexingLatencyP99Alarm: {
        Warning: {
          maxLatency: Duration.millis(700),
        },
      },
      addSearchLatencyP50Alarm: {
        Warning: {
          maxLatency: Duration.millis(500),
        },
      },
      addSearchLatencyP90Alarm: {
        Warning: {
          maxLatency: Duration.millis(600),
        },
      },
      addSearchLatencyP99Alarm: {
        Warning: {
          maxLatency: Duration.millis(700),
        },
      },
      addClusterStatusAlarm: {
        Warning: {
          status: OpenSearchClusterStatus.YELLOW,
        },
        Critical: {
          status: OpenSearchClusterStatus.RED,
        },
      },
      addDiskSpaceUsageAlarm: {
        Warning: {
          maxUsagePercent: 80,
        },
      },
      addCpuSpaceUsageAlarm: {
        Warning: {
          maxUsagePercent: 80,
        },
      },
      addMasterCpuSpaceUsageAlarm: {
        Warning: {
          maxUsagePercent: 80,
        },
      },
      addJvmMemoryPressureAlarm: {
        Warning: {
          maxUsagePercent: 80,
        },
      },
      addMasterJvmMemoryPressureAlarm: {
        Warning: {
          maxUsagePercent: 80,
        },
      },
      addClusterIndexWritesBlockedAlarm: {
        Warning: {
          maxBlockedWrites: 1,
        },
      },
      addClusterNodeCountAlarm: {
        Warning: {
          minNodes: 1,
        },
      },
      addClusterAutomatedSnapshotFailureAlarm: {
        Warning: {
          maxFailures: 1,
        },
      },
      addKmsKeyErrorAlarm: {
        Warning: {
          maxErrors: 1,
        },
      },
      addKmsKeyInaccessibleAlarm: {
        Warning: {
          maxAccessAttempts: 1,
        },
      },
      useCreatedAlarms: {
        consume(alarms: AlarmWithAnnotation[]) {
          numAlarmsCreated = alarms.length;
        },
      },
    });

    expect(numAlarmsCreated).toStrictEqual(18);
    expect(Template.fromStack(stack)).toMatchSnapshot();
  });
});
