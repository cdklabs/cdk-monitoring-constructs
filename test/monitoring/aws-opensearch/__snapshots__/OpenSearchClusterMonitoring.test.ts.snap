// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: all alarms for ElasticsearchCfnDomain 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "ElasticsearchCfnDomain": Object {
      "Properties": Object {
        "DomainName": "es-cfn-domain",
      },
      "Type": "AWS::Elasticsearch::Domain",
    },
    "ElasticsearchDomain54566C07": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "CognitoOptions": Object {
          "Enabled": false,
        },
        "DomainEndpointOptions": Object {
          "EnforceHTTPS": false,
          "TLSSecurityPolicy": "Policy-Min-TLS-1-0-2019-07",
        },
        "DomainName": "es-domain",
        "EBSOptions": Object {
          "EBSEnabled": true,
          "VolumeSize": 10,
          "VolumeType": "gp2",
        },
        "ElasticsearchClusterConfig": Object {
          "DedicatedMasterEnabled": false,
          "InstanceCount": 1,
          "InstanceType": "r5.large.elasticsearch",
          "ZoneAwarenessEnabled": false,
        },
        "ElasticsearchVersion": "7.7",
        "EncryptionAtRestOptions": Object {
          "Enabled": false,
        },
        "LogPublishingOptions": Object {},
        "NodeToNodeEncryptionOptions": Object {
          "Enabled": false,
        },
      },
      "Type": "AWS::Elasticsearch::Domain",
      "UpdateReplacePolicy": "Retain",
    },
    "OpenSearchCfnDomain": Object {
      "Properties": Object {
        "DomainName": "os-cfn-domain",
      },
      "Type": "AWS::OpenSearchService::Domain",
    },
    "OpenSearchDomain85D65221": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "ClusterConfig": Object {
          "DedicatedMasterEnabled": false,
          "InstanceCount": 1,
          "InstanceType": "r5.large.search",
          "ZoneAwarenessEnabled": false,
        },
        "DomainEndpointOptions": Object {
          "EnforceHTTPS": false,
          "TLSSecurityPolicy": "Policy-Min-TLS-1-0-2019-07",
        },
        "DomainName": "os-domain",
        "EBSOptions": Object {
          "EBSEnabled": true,
          "VolumeSize": 10,
          "VolumeType": "gp2",
        },
        "EncryptionAtRestOptions": Object {
          "Enabled": false,
        },
        "EngineVersion": "OpenSearch_1.0",
        "LogPublishingOptions": Object {},
        "NodeToNodeEncryptionOptions": Object {
          "Enabled": false,
        },
      },
      "Type": "AWS::OpenSearchService::Domain",
      "UpdateReplacePolicy": "Retain",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;

exports[`snapshot test: all alarms for ElasticsearchDomain 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "ElasticsearchCfnDomain": Object {
      "Properties": Object {
        "DomainName": "es-cfn-domain",
      },
      "Type": "AWS::Elasticsearch::Domain",
    },
    "ElasticsearchDomain54566C07": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "CognitoOptions": Object {
          "Enabled": false,
        },
        "DomainEndpointOptions": Object {
          "EnforceHTTPS": false,
          "TLSSecurityPolicy": "Policy-Min-TLS-1-0-2019-07",
        },
        "DomainName": "es-domain",
        "EBSOptions": Object {
          "EBSEnabled": true,
          "VolumeSize": 10,
          "VolumeType": "gp2",
        },
        "ElasticsearchClusterConfig": Object {
          "DedicatedMasterEnabled": false,
          "InstanceCount": 1,
          "InstanceType": "r5.large.elasticsearch",
          "ZoneAwarenessEnabled": false,
        },
        "ElasticsearchVersion": "7.7",
        "EncryptionAtRestOptions": Object {
          "Enabled": false,
        },
        "LogPublishingOptions": Object {},
        "NodeToNodeEncryptionOptions": Object {
          "Enabled": false,
        },
      },
      "Type": "AWS::Elasticsearch::Domain",
      "UpdateReplacePolicy": "Retain",
    },
    "OpenSearchCfnDomain": Object {
      "Properties": Object {
        "DomainName": "os-cfn-domain",
      },
      "Type": "AWS::OpenSearchService::Domain",
    },
    "OpenSearchDomain85D65221": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "ClusterConfig": Object {
          "DedicatedMasterEnabled": false,
          "InstanceCount": 1,
          "InstanceType": "r5.large.search",
          "ZoneAwarenessEnabled": false,
        },
        "DomainEndpointOptions": Object {
          "EnforceHTTPS": false,
          "TLSSecurityPolicy": "Policy-Min-TLS-1-0-2019-07",
        },
        "DomainName": "os-domain",
        "EBSOptions": Object {
          "EBSEnabled": true,
          "VolumeSize": 10,
          "VolumeType": "gp2",
        },
        "EncryptionAtRestOptions": Object {
          "Enabled": false,
        },
        "EngineVersion": "OpenSearch_1.0",
        "LogPublishingOptions": Object {},
        "NodeToNodeEncryptionOptions": Object {
          "Enabled": false,
        },
      },
      "Type": "AWS::OpenSearchService::Domain",
      "UpdateReplacePolicy": "Retain",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;

exports[`snapshot test: all alarms for OpenSearchCfnDomain 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "ElasticsearchCfnDomain": Object {
      "Properties": Object {
        "DomainName": "es-cfn-domain",
      },
      "Type": "AWS::Elasticsearch::Domain",
    },
    "ElasticsearchDomain54566C07": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "CognitoOptions": Object {
          "Enabled": false,
        },
        "DomainEndpointOptions": Object {
          "EnforceHTTPS": false,
          "TLSSecurityPolicy": "Policy-Min-TLS-1-0-2019-07",
        },
        "DomainName": "es-domain",
        "EBSOptions": Object {
          "EBSEnabled": true,
          "VolumeSize": 10,
          "VolumeType": "gp2",
        },
        "ElasticsearchClusterConfig": Object {
          "DedicatedMasterEnabled": false,
          "InstanceCount": 1,
          "InstanceType": "r5.large.elasticsearch",
          "ZoneAwarenessEnabled": false,
        },
        "ElasticsearchVersion": "7.7",
        "EncryptionAtRestOptions": Object {
          "Enabled": false,
        },
        "LogPublishingOptions": Object {},
        "NodeToNodeEncryptionOptions": Object {
          "Enabled": false,
        },
      },
      "Type": "AWS::Elasticsearch::Domain",
      "UpdateReplacePolicy": "Retain",
    },
    "OpenSearchCfnDomain": Object {
      "Properties": Object {
        "DomainName": "os-cfn-domain",
      },
      "Type": "AWS::OpenSearchService::Domain",
    },
    "OpenSearchDomain85D65221": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "ClusterConfig": Object {
          "DedicatedMasterEnabled": false,
          "InstanceCount": 1,
          "InstanceType": "r5.large.search",
          "ZoneAwarenessEnabled": false,
        },
        "DomainEndpointOptions": Object {
          "EnforceHTTPS": false,
          "TLSSecurityPolicy": "Policy-Min-TLS-1-0-2019-07",
        },
        "DomainName": "os-domain",
        "EBSOptions": Object {
          "EBSEnabled": true,
          "VolumeSize": 10,
          "VolumeType": "gp2",
        },
        "EncryptionAtRestOptions": Object {
          "Enabled": false,
        },
        "EngineVersion": "OpenSearch_1.0",
        "LogPublishingOptions": Object {},
        "NodeToNodeEncryptionOptions": Object {
          "Enabled": false,
        },
      },
      "Type": "AWS::OpenSearchService::Domain",
      "UpdateReplacePolicy": "Retain",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;

exports[`snapshot test: all alarms for OpenSearchDomain 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "ElasticsearchCfnDomain": Object {
      "Properties": Object {
        "DomainName": "es-cfn-domain",
      },
      "Type": "AWS::Elasticsearch::Domain",
    },
    "ElasticsearchDomain54566C07": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "CognitoOptions": Object {
          "Enabled": false,
        },
        "DomainEndpointOptions": Object {
          "EnforceHTTPS": false,
          "TLSSecurityPolicy": "Policy-Min-TLS-1-0-2019-07",
        },
        "DomainName": "es-domain",
        "EBSOptions": Object {
          "EBSEnabled": true,
          "VolumeSize": 10,
          "VolumeType": "gp2",
        },
        "ElasticsearchClusterConfig": Object {
          "DedicatedMasterEnabled": false,
          "InstanceCount": 1,
          "InstanceType": "r5.large.elasticsearch",
          "ZoneAwarenessEnabled": false,
        },
        "ElasticsearchVersion": "7.7",
        "EncryptionAtRestOptions": Object {
          "Enabled": false,
        },
        "LogPublishingOptions": Object {},
        "NodeToNodeEncryptionOptions": Object {
          "Enabled": false,
        },
      },
      "Type": "AWS::Elasticsearch::Domain",
      "UpdateReplacePolicy": "Retain",
    },
    "OpenSearchCfnDomain": Object {
      "Properties": Object {
        "DomainName": "os-cfn-domain",
      },
      "Type": "AWS::OpenSearchService::Domain",
    },
    "OpenSearchDomain85D65221": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "ClusterConfig": Object {
          "DedicatedMasterEnabled": false,
          "InstanceCount": 1,
          "InstanceType": "r5.large.search",
          "ZoneAwarenessEnabled": false,
        },
        "DomainEndpointOptions": Object {
          "EnforceHTTPS": false,
          "TLSSecurityPolicy": "Policy-Min-TLS-1-0-2019-07",
        },
        "DomainName": "os-domain",
        "EBSOptions": Object {
          "EBSEnabled": true,
          "VolumeSize": 10,
          "VolumeType": "gp2",
        },
        "EncryptionAtRestOptions": Object {
          "Enabled": false,
        },
        "EngineVersion": "OpenSearch_1.0",
        "LogPublishingOptions": Object {},
        "NodeToNodeEncryptionOptions": Object {
          "Enabled": false,
        },
      },
      "Type": "AWS::OpenSearchService::Domain",
      "UpdateReplacePolicy": "Retain",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;

exports[`snapshot test: no alarms for ElasticsearchCfnDomain 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "ElasticsearchCfnDomain": Object {
      "Properties": Object {
        "DomainName": "es-cfn-domain",
      },
      "Type": "AWS::Elasticsearch::Domain",
    },
    "ElasticsearchDomain54566C07": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "CognitoOptions": Object {
          "Enabled": false,
        },
        "DomainEndpointOptions": Object {
          "EnforceHTTPS": false,
          "TLSSecurityPolicy": "Policy-Min-TLS-1-0-2019-07",
        },
        "DomainName": "es-domain",
        "EBSOptions": Object {
          "EBSEnabled": true,
          "VolumeSize": 10,
          "VolumeType": "gp2",
        },
        "ElasticsearchClusterConfig": Object {
          "DedicatedMasterEnabled": false,
          "InstanceCount": 1,
          "InstanceType": "r5.large.elasticsearch",
          "ZoneAwarenessEnabled": false,
        },
        "ElasticsearchVersion": "7.7",
        "EncryptionAtRestOptions": Object {
          "Enabled": false,
        },
        "LogPublishingOptions": Object {},
        "NodeToNodeEncryptionOptions": Object {
          "Enabled": false,
        },
      },
      "Type": "AWS::Elasticsearch::Domain",
      "UpdateReplacePolicy": "Retain",
    },
    "OpenSearchCfnDomain": Object {
      "Properties": Object {
        "DomainName": "os-cfn-domain",
      },
      "Type": "AWS::OpenSearchService::Domain",
    },
    "OpenSearchDomain85D65221": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "ClusterConfig": Object {
          "DedicatedMasterEnabled": false,
          "InstanceCount": 1,
          "InstanceType": "r5.large.search",
          "ZoneAwarenessEnabled": false,
        },
        "DomainEndpointOptions": Object {
          "EnforceHTTPS": false,
          "TLSSecurityPolicy": "Policy-Min-TLS-1-0-2019-07",
        },
        "DomainName": "os-domain",
        "EBSOptions": Object {
          "EBSEnabled": true,
          "VolumeSize": 10,
          "VolumeType": "gp2",
        },
        "EncryptionAtRestOptions": Object {
          "Enabled": false,
        },
        "EngineVersion": "OpenSearch_1.0",
        "LogPublishingOptions": Object {},
        "NodeToNodeEncryptionOptions": Object {
          "Enabled": false,
        },
      },
      "Type": "AWS::OpenSearchService::Domain",
      "UpdateReplacePolicy": "Retain",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;

exports[`snapshot test: no alarms for ElasticsearchDomain 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "ElasticsearchCfnDomain": Object {
      "Properties": Object {
        "DomainName": "es-cfn-domain",
      },
      "Type": "AWS::Elasticsearch::Domain",
    },
    "ElasticsearchDomain54566C07": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "CognitoOptions": Object {
          "Enabled": false,
        },
        "DomainEndpointOptions": Object {
          "EnforceHTTPS": false,
          "TLSSecurityPolicy": "Policy-Min-TLS-1-0-2019-07",
        },
        "DomainName": "es-domain",
        "EBSOptions": Object {
          "EBSEnabled": true,
          "VolumeSize": 10,
          "VolumeType": "gp2",
        },
        "ElasticsearchClusterConfig": Object {
          "DedicatedMasterEnabled": false,
          "InstanceCount": 1,
          "InstanceType": "r5.large.elasticsearch",
          "ZoneAwarenessEnabled": false,
        },
        "ElasticsearchVersion": "7.7",
        "EncryptionAtRestOptions": Object {
          "Enabled": false,
        },
        "LogPublishingOptions": Object {},
        "NodeToNodeEncryptionOptions": Object {
          "Enabled": false,
        },
      },
      "Type": "AWS::Elasticsearch::Domain",
      "UpdateReplacePolicy": "Retain",
    },
    "OpenSearchCfnDomain": Object {
      "Properties": Object {
        "DomainName": "os-cfn-domain",
      },
      "Type": "AWS::OpenSearchService::Domain",
    },
    "OpenSearchDomain85D65221": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "ClusterConfig": Object {
          "DedicatedMasterEnabled": false,
          "InstanceCount": 1,
          "InstanceType": "r5.large.search",
          "ZoneAwarenessEnabled": false,
        },
        "DomainEndpointOptions": Object {
          "EnforceHTTPS": false,
          "TLSSecurityPolicy": "Policy-Min-TLS-1-0-2019-07",
        },
        "DomainName": "os-domain",
        "EBSOptions": Object {
          "EBSEnabled": true,
          "VolumeSize": 10,
          "VolumeType": "gp2",
        },
        "EncryptionAtRestOptions": Object {
          "Enabled": false,
        },
        "EngineVersion": "OpenSearch_1.0",
        "LogPublishingOptions": Object {},
        "NodeToNodeEncryptionOptions": Object {
          "Enabled": false,
        },
      },
      "Type": "AWS::OpenSearchService::Domain",
      "UpdateReplacePolicy": "Retain",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;

exports[`snapshot test: no alarms for OpenSearchCfnDomain 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "ElasticsearchCfnDomain": Object {
      "Properties": Object {
        "DomainName": "es-cfn-domain",
      },
      "Type": "AWS::Elasticsearch::Domain",
    },
    "ElasticsearchDomain54566C07": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "CognitoOptions": Object {
          "Enabled": false,
        },
        "DomainEndpointOptions": Object {
          "EnforceHTTPS": false,
          "TLSSecurityPolicy": "Policy-Min-TLS-1-0-2019-07",
        },
        "DomainName": "es-domain",
        "EBSOptions": Object {
          "EBSEnabled": true,
          "VolumeSize": 10,
          "VolumeType": "gp2",
        },
        "ElasticsearchClusterConfig": Object {
          "DedicatedMasterEnabled": false,
          "InstanceCount": 1,
          "InstanceType": "r5.large.elasticsearch",
          "ZoneAwarenessEnabled": false,
        },
        "ElasticsearchVersion": "7.7",
        "EncryptionAtRestOptions": Object {
          "Enabled": false,
        },
        "LogPublishingOptions": Object {},
        "NodeToNodeEncryptionOptions": Object {
          "Enabled": false,
        },
      },
      "Type": "AWS::Elasticsearch::Domain",
      "UpdateReplacePolicy": "Retain",
    },
    "OpenSearchCfnDomain": Object {
      "Properties": Object {
        "DomainName": "os-cfn-domain",
      },
      "Type": "AWS::OpenSearchService::Domain",
    },
    "OpenSearchDomain85D65221": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "ClusterConfig": Object {
          "DedicatedMasterEnabled": false,
          "InstanceCount": 1,
          "InstanceType": "r5.large.search",
          "ZoneAwarenessEnabled": false,
        },
        "DomainEndpointOptions": Object {
          "EnforceHTTPS": false,
          "TLSSecurityPolicy": "Policy-Min-TLS-1-0-2019-07",
        },
        "DomainName": "os-domain",
        "EBSOptions": Object {
          "EBSEnabled": true,
          "VolumeSize": 10,
          "VolumeType": "gp2",
        },
        "EncryptionAtRestOptions": Object {
          "Enabled": false,
        },
        "EngineVersion": "OpenSearch_1.0",
        "LogPublishingOptions": Object {},
        "NodeToNodeEncryptionOptions": Object {
          "Enabled": false,
        },
      },
      "Type": "AWS::OpenSearchService::Domain",
      "UpdateReplacePolicy": "Retain",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;

exports[`snapshot test: no alarms for OpenSearchDomain 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "ElasticsearchCfnDomain": Object {
      "Properties": Object {
        "DomainName": "es-cfn-domain",
      },
      "Type": "AWS::Elasticsearch::Domain",
    },
    "ElasticsearchDomain54566C07": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "CognitoOptions": Object {
          "Enabled": false,
        },
        "DomainEndpointOptions": Object {
          "EnforceHTTPS": false,
          "TLSSecurityPolicy": "Policy-Min-TLS-1-0-2019-07",
        },
        "DomainName": "es-domain",
        "EBSOptions": Object {
          "EBSEnabled": true,
          "VolumeSize": 10,
          "VolumeType": "gp2",
        },
        "ElasticsearchClusterConfig": Object {
          "DedicatedMasterEnabled": false,
          "InstanceCount": 1,
          "InstanceType": "r5.large.elasticsearch",
          "ZoneAwarenessEnabled": false,
        },
        "ElasticsearchVersion": "7.7",
        "EncryptionAtRestOptions": Object {
          "Enabled": false,
        },
        "LogPublishingOptions": Object {},
        "NodeToNodeEncryptionOptions": Object {
          "Enabled": false,
        },
      },
      "Type": "AWS::Elasticsearch::Domain",
      "UpdateReplacePolicy": "Retain",
    },
    "OpenSearchCfnDomain": Object {
      "Properties": Object {
        "DomainName": "os-cfn-domain",
      },
      "Type": "AWS::OpenSearchService::Domain",
    },
    "OpenSearchDomain85D65221": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "ClusterConfig": Object {
          "DedicatedMasterEnabled": false,
          "InstanceCount": 1,
          "InstanceType": "r5.large.search",
          "ZoneAwarenessEnabled": false,
        },
        "DomainEndpointOptions": Object {
          "EnforceHTTPS": false,
          "TLSSecurityPolicy": "Policy-Min-TLS-1-0-2019-07",
        },
        "DomainName": "os-domain",
        "EBSOptions": Object {
          "EBSEnabled": true,
          "VolumeSize": 10,
          "VolumeType": "gp2",
        },
        "EncryptionAtRestOptions": Object {
          "Enabled": false,
        },
        "EngineVersion": "OpenSearch_1.0",
        "LogPublishingOptions": Object {},
        "NodeToNodeEncryptionOptions": Object {
          "Enabled": false,
        },
      },
      "Type": "AWS::OpenSearchService::Domain",
      "UpdateReplacePolicy": "Retain",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
